test/framework/src/main/java/org/elasticsearch/repositories/AbstractThirdPartyRepositoryTestCase.java

```diff
@@ -309,6 +316,39 @@ public void testReadFromPositionWithLength() {
         }
     }
 
+    protected void testReadFromPositionLargerThanBlobLength(Predicate<RequestedRangeNotSatisfiedException> responseCodeChecker) {
+        final var blobName = randomIdentifier();
+        final var blobBytes = randomBytesReference(randomIntBetween(100, 2_000));
+
+        final var repository = getRepository();
+        executeOnBlobStore(repository, blobStore -> {
+            blobStore.writeBlob(randomPurpose(), blobName, blobBytes, true);
+            return null;
+        });
+
+        long position = randomLongBetween(blobBytes.length(), Long.MAX_VALUE - 1L);
+        long length = randomLongBetween(1L, Long.MAX_VALUE - position);
+
+        var exception = expectThrows(UncategorizedExecutionException.class, () -> readBlob(repository, blobName, position, length));
+        assertThat(exception.getCause(), instanceOf(ExecutionException.class));
+        assertThat(exception.getCause().getCause(), instanceOf(RequestedRangeNotSatisfiedException.class));
+        assertThat(
```

Comment:
nit: can we check explicitely RequestedRangeNotSatisfiedException#getPosition and getLength?